### ISSUE TITLE
0.46.1- MekHQ counts vehicles destroyed by Tornadoes as Undamaged at Scenario End

### ISSUE THREAD

**Issue by Windchild292** [MegaMek/mekhq#1498]
The vehicles on this list were all destroyed by a tornado, yet show as undamaged in MekHQ.
![image](https://user-images.githubusercontent.com/39067288/74872394-26549d00-532b-11ea-85ca-dd9b20a83dda.png)

**Comment by RAldrich** [comment]
I actually don't think showing them as damaged is the correct behavior. Tactical Operations (pg 62) just says: "Non-’Mech Units: These units cannot operate in Tornado F4+ conditions."

This is similar to infantry in extreme temperatures - they're not supposed to die instantly, you're just not supposed to be able to deploy them in the first place (unless they're in a building, but that's not implemented).

"F4 Tornado" would probably have better been called a "Hurricane", but even if you think of it as a literal F4 Tornado, I don't think it would auto-kill a heavy tank. I think removing them from the game represents the fact that they are buttoned up and tactically out of action. They're probably fine after the storm passes.

**Comment by rroby1** [comment]
Maybe not a heavy tank, but the Pegasus is a light at around 35 tons. Well within the range that an F4 tornado can throw. I have seen reports that 35.8 ton coal cars have been thrown 391 feet. This would certainly cause damage. Other damage reported include 95 ton oil storage tanks being ripped from their foundations and rolled several miles.  That is well within the range of  the 50 ton Vendette and Maxim and 60 ton Bulldog. Again some sort of damage would occur. May not be enough to make them combat ineffective, but there should be some damage.    Auto-kill, no. 

I've had mechs become inoperable just from maintenance. I would expect at least some light damage from a tornado, especially an F4 / F5. Tracked or wheeled vehicles could be immobilized due to damage to the tires / tracks. Sensors damaged by flying debris damaging the external components. Same for comms systems having their antenna damaged. So, there is all kinds of light damage which can happen. Really don't see an issue with allowing a small chance at some damage, even as a option.

**Comment by HammerGS** [comment]
As mentioned I think the bigger issues is ""Non-’Mech Units: These units cannot operate in Tornado F4+ conditions.""

So why are we letting them deploy. It's feeling like an easy cheat for the player 6 tanks worth of salvage without firing a shot.  My .02 cents.

**Comment by rroby1** [comment]
Very true. I haven't played around with weather. Too much of a hassle on the table top trying to remember all the modifiers. I've not used in in MM yet. Can the weather change during a scenario?

**Comment by RAldrich** [comment]
> Very true. I haven't played around with weather. Too much of a hassle on the table top trying to remember all the modifiers. I've not used in in MM yet. Can the weather change during a scenario?

Wind can change speed and direction, which can result in a Tornado - this is off by default and can be configured in MegaMek. Other weather cannot change during a mission.

>  Really don't see an issue with allowing a small chance at some damage, even as a option.

I agree, with the caveat that it's not anywhere in the rules, so it should be optional (unofficial).

Really Battletech's weather system makes NO sense. For example, rules as written, light snowfall forces the temperature to extreme cold, -50 degrees Celsius. That is ABSURDLY cold. And there's a variety of interactions with what's allowed given a particular atmosphere density, and there's some bugs with AtB rules generating weather that's not permitted, and so on.

**Comment by RAldrich** [comment]
> As mentioned I think the bigger issues is ""Non-’Mech Units: These units cannot operate in Tornado F4+ conditions.""
> 
> So why are we letting them deploy. It's feeling like an easy cheat for the player 6 tanks worth of salvage without firing a shot. My .02 cents.

I agree. I don't think MegaMek has the full Tornado rules implemented. Both "infantry in buildings should be OK" and "mechs have to crawl everywhere" don't happen as far as I know.

Maybe a relatively easy fix would be preventing AtB from selecting vehicles and infantry during scenario generation when the weather doesn't allow their deployment?

**Comment by Tzahr** [comment]
I observed the following: 

Units will deploy in conditions they cannot survive, and be destroyed, at least if the plantery conditions are manually altered to tornado. 

Those units will show up fully intact in salvage if board control is declared. 

This is current as of MekHQ revision: 042c1936d192f60d5cc36fd50dd55301423ccee4
